### PR TITLE
[Order Creation M2] API prices calculation

### LINKED ISSUE #5693: [Order Creation] Delegate price calculation to the Store API
Before starting further development around the Payments section of the Order Creation feature, we need to delegate some of the price calculations of an Order through the draft status we now have available at the merchant store API. This task should be the foundation for the entire Order Creation project M2 deliverable.

### PR DESCRIPTION
**Pull request by hichamboushaba**
⚠️ Please don't merge until the FluxC [PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2246) is reviewed and merged

### Description
This PR implements order draft creation and syncing to get the order prices from the API and display the total taxes, and will be used as a base foundation for the shipping and fees later.

All changes are behind a feature flag `ORDER_CREATION_M2`, to allow merging this without breaking the current logic for M1.

##### Test API prices calculation
Please make sure your store defines some tax rates (you just need two different Standard rates for two different regions, check this [doc](https://woocommerce.com/document/setting-up-taxes-in-woocommerce) for more details if needed)
Then create some orders on the app and compare the taxes calculation to orders created through wp-admin, and confirm they match.

Please repeat the test with the option ["Prices Entered With Tax"](https://woocommerce.com/document/setting-up-taxes-in-woocommerce/#prices-entered-with-tax) set to yes.

##### Non-regression for M1
Please disable manually the `ORDER_CREATION_M2` flag, and make sure all of the new changes are hidden, and that order creation works as expected.

##### Updating products
https://user-images.githubusercontent.com/1657201/150776531-1764836c-1b43-4486-a1ba-274a15ea25da.mp4

##### Updaging shipping address
https://user-images.githubusercontent.com/1657201/150776543-27f74b15-6ca9-428c-a960-5f16c46b5eca.mp4

##### Error state
https://user-images.githubusercontent.com/1657201/150776557-ac376d9e-41f0-48b9-86a3-fe96fa1dd80b.mov

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreationFormFragment.kt [hunk 1]
```diff
@@ -273,6 +285,27 @@ class OrderCreationFormFragment : BaseFragment(R.layout.fragment_order_creation_
         progressDialog?.isCancelable = false
     }
 
+    @Suppress("MagicNumber")
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreationFormFragment.kt:288]: I don't think this delay value needs a constant, having ton of constants reduces readability sometimes, but let me know if you don't agree.

File: WooCommerce/src/test/kotlin/com/woocommerce/android/util/InlineClassesAnswer.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import org.mockito.invocation.InvocationOnMock
+import org.mockito.stubbing.Answer
+
+/**
+ * A utility Mockito [Answer] to support kotlin inline classes.
+ * It uses internally the same functionality that Mockito uses for supporting inline classes when using [Returns]
+ *
+ * check https://github.com/mockito/mockito/pull/2280
+ */
+class InlineClassesAnswer<T : Any>(private val defaultAnswer: Answer<T>) : Answer<T> {
```
Comment (hichamboushaba) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/util/InlineClassesAnswer.kt:14]: This is useful for cases where the return type of our mock uses an inline class, like `kotlin.Result`.

File: WooCommerce/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -350,12 +350,13 @@
     <string name="order_creation_payment">Payment</string>
     <string name="order_creation_payment_products_total">Products Total</string>
     <string name="order_creation_payment_order_total">Order Total</string>
-    <string name="order_creation_payment_tax_hint">Total excluding taxes. Taxes will be automatically calculated based on your store settings.</string>
+    <string name="order_creation_payment_tax_hint">Taxes will be automatically calculated based on your store settings.</string>
```
Comment (hichamboushaba) [inline WooCommerce/src/main/res/values/strings.xml:353]: internal discussion: p1643131391027900-slack-C02KUCFCSFP

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreationFormFragment.kt (hunk 1 above)
Comment (ThomazFB) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreationFormFragment.kt:288]: I agree, no need to give names to simple time measures.

**Review by ThomazFB — APPROVED** [review]
Reviewing this PR was a masterclass on flow usage, @hichamboushaba. I learned a lot with the `CreateOrUpdateOrderDraft` use case and how you connected the stream of status changes with the ViewModel. Impressive work here. Thank you a lot for all the effort. I'm approving the PR and leaving to you the FluxC merging sync.

**Pull request merged**